fv_be/app/models/sites.py

```diff
@@ -0,0 +1,133 @@
+from django.core.validators import validate_slug
+from django.db import models
+
+from fv_be.app.models import BaseModel
+from fv_be.users.models import User
+
+from .constants import Role, Visibility
+
+
+class LanguageFamily(BaseModel):
+    """
+    Represents a Language Family.
+    """
+
+    # from fvlanguage type
+
+    class Meta:
+        # todo: i18n
+        verbose_name = "language family"
+        verbose_name_plural = "language families"
+
+    # from dc:title
+    title = models.CharField(max_length=200)
+
+    def __str__(self):
+        return self.title
+
+
+class Site(BaseModel):
+    """
+    Represents Language Site and contains the basic public information about the site. For site information that should
+    be access-controlled, see the SiteInformation model.
+    """
+
+    # from fvdialect type
+
+    # from dc:title
+    title = models.CharField(max_length=200)
+
+    # from fvdialect:short_url
+    slug = models.SlugField(
+        max_length=200, blank=False, validators=[validate_slug], db_index=True
+    )
+
+    # from fva:language
+    language_family = models.ForeignKey(LanguageFamily, on_delete=models.DO_NOTHING)
+
+    # from state (will have to be translated from existing states to new visibilities)
+    visibility = models.IntegerField(
+        choices=Visibility.choices, default=Visibility.TEAM, db_index=True
+    )
+
+    # todo: add logo field when media is ready / from fvdialect:logo
+    # todo: add banner field when media is ready / from fvdialect:background_top_image
+
+    def __str__(self):
+        return self.title
+
+
+class BaseSiteContentModel(BaseModel):
+    """
+    Base model for non-access-controlled site content data such as categories, that do not have their own
+    visibility levels. Can also be used as a base for more specific types of site content base models.
+    """
+
+    class Meta:
+        abstract = True
+
+    site = models.ForeignKey(Site, on_delete=models.CASCADE, db_index=True)
+
+
+class BaseControlledSiteContentModel(BaseSiteContentModel):
+    """
+    Base model for access-controlled site content models such as words, phrases, songs, and stories, that have their own
+    visibility level setting.
+    """
+
+    class Meta:
+        abstract = True
+
+    visibility = models.IntegerField(
+        choices=Visibility.choices, default=Visibility.TEAM
+    )
+
+
+class SiteInformation(BaseSiteContentModel):
+    """Contains the access-controlled information about a language site."""
+
+    # details come from fvdialect document
+
```

Comment:
AFAIK the following fields are all related to the V1 "portal" and are not leveraged in V2:
- fvdialect:contact_information
- fvdialect:greeting
- fvdialect:about_our_language
- fvdialect:about_us
- dc:description
- fvdialect:featured_words
- fvdialect:featured_audio
- fvdialect:keyboards
In V2 these are handled with widgets, so rather than migrate and store them here, I would suggest migrating them as Widgets in whichever widget is the best fit, and dropping them otherwise. -- @gmcauliffe ?